src/SignalR/server/Core/src/DefaultHubLifetimeManager.cs

```diff
@@ -314,4 +325,63 @@ public override Task SendUsersAsync(IReadOnlyList<string> userIds, string method
     {
         return SendToAllConnections(methodName, args, (connection, state) => ((IReadOnlyList<string>)state!).Contains(connection.UserIdentifier), userIds, cancellationToken);
     }
+
+    /// <inheritdoc/>
+    public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object?[] args, CancellationToken cancellationToken = default)
+    {
+        if (connectionId == null)
+        {
+            throw new ArgumentNullException(nameof(connectionId));
+        }
+
+        var connection = _connections[connectionId];
+
+        if (connection == null)
+        {
+            throw new InvalidOperationException("Connection does not exist.");
+        }
+
+        var invocationId = Interlocked.Increment(ref _lastInvocationId).ToString(NumberFormatInfo.InvariantInfo);
+        var task = _clientResultsManager.AddInvocation<T>(connectionId, invocationId, cancellationToken);
+        // Connection disconnected while adding invocation
+        // we need to try to remove it here to avoid the task hanging if the add happened after the connection cleanup
+        if (connection.ConnectionAborted.IsCancellationRequested)
```

Comment:
This is an optimization right? It could be cancelled right after this check as well.

Reply:
This is to fix the race where a disconnect and an invocation are happening at the same time. Disconnect cleans up any active invocations, but because one could be added right after the cleanup we need to check if the connection is closed after adding it to avoid the invocation being added and never removed.

Reply:
Notes: we can wire up a linked cts and catch OCE to preserve the "Connection disconnected" error message. Would fix the race and allow us to remove the cleanup code in OnDisconnectedAsync